scheduler/src/cook/mesos/util.clj

```diff
@@ -363,22 +413,30 @@
                                    (-> 1 t/hours t/in-millis)))
           expanded-end (Date. (+ (.getTime end)
                                  (-> 1 t/hours t/in-millis)))
+          entid-start (d/entid-at db :db.part/user expanded-start)
+          entid-end (d/entid-at db :db.part/user expanded-end)
+          job-user-entid (d/entid db :job/user)
+          ^long start-ms (.getTime start)
+          ^long end-ms (.getTime end)
           jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
+          (->> (d/seek-datoms db :avet :job/user user entid-start)
+               (take-while #(and (< (:e %) entid-end)
+                                 (= (:a %) job-user-entid)
+                                 (= (:v %) user)))
+               (map :e)
                (map (partial d/entity db))
-               (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
-               (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
+               (filter #(let [^long submit-ms (.getTime (:job/submit-time %))]
+                          (and (<= start-ms submit-ms)
+                               (< submit-ms end-ms))))
```

Comment:
Seems like it would be useful to `defn` this function since we use it twice, e.g. `(submitted-in-range? start end job)`

Reply:
Done